File: CODIGOS6/104.kt
fun main(args: Array<String>) {
  val n = readLine()!!.toInt()
  val eggs = IntArray(20)
  val touching = Array(20) { IntArray(20) }
  val ans = mutableListOf<Int>()
  for (i in 0 until n) {
    cin >> eggs[i]
  }
  for (i in 0 until n) {
    for (j in 0 until n) {
      touching[i][j] = 0
    }
  }
  for (i in 0 until n) {
    val k = readLine()!!.toInt()
    for (j in 0 until k) {
      val t = readLine()!!.toInt()
      touching[i][t] = 1
    }
  }
  var cnt = 0
  for (i in 0 until n) {
    var flag = true
    for (j in 0 until n) {
      if (eggs[j] == 1 && touching[i][j] == 1) {
        flag = false
        break
      }
    }
    if (flag) {
      ans.add(i)
    }
  }
  for (i in 0 until ans.size) {
    print(ans[i] + " ")
  }
  println()
}

File: CODIGOS6/104.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
  int n, eggs[20], touching[20][20], ans[20];
  cin >> n;
  for (int i = 0; i < n; i++) {
    cin >> eggs[i];
  }
  for (int i = 0; i < n; i++) {
    for (int j = 0; j < n; j++) {
      touching[i][j] = 0;
    }
  }
  for (int i = 0; i < n; i++) {
    int k;
    cin >> k;
    for (int j = 0; j < k; j++) {
      int t;
      cin >> t;
      touching[i][t] = 1;
    }
  }
  int cnt = 0;
  for (int i = 0; i < n; i++) {
    int flag = 1;
    for (int j = 0; j < n; j++) {
      if (eggs[j] == 1 && touching[i][j] == 1) {
        flag = 0;
        break;
      }
    }
    if (flag) {
      ans[cnt++] = i;
    }
  }
  for (int i = 0; i < cnt; i++) {
    cout << ans[i] << " ";
  }
  cout << endl;
  return 0;
}
